endpoint.go

```diff
@@ -303,14 +306,16 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) (*Contai
 		}
 	}
 
-	err = sb.SetGateway(joinInfo.gw)
-	if err != nil {
-		return nil, err
-	}
+	if primary {
```

Comment:
where do we fail ? iirc `sb.SetGateway` doesn't fail for if gw is overwritten.
regardless, having a `primary` bool in endpoint doesnt seem to be appropriate for the use-case that this PR is trying to address.
Since we are trying to identify the default-gateway for a container with multiple endpoints, I believe it is more appropriate for the sandbox to hold the info on which is the primary endpoint. That's because, as per design a container might join or leave an endpoint dynamically. sandbox need not be a permanent back-end for the endpoint (or) vice-versa. Hence, endpoint carrying this info may not be appropriate. wdyt ?